src/nvim/ui_compositor.c

```diff
@@ -334,17 +334,31 @@ static void compose_line(Integer row, Integer startcol, Integer endcol,
   sattr_T *bg_attrs = &default_grid.attrs[default_grid.line_offset[row]
                                           +(size_t)startcol];
 
+  int grid_rows, grid_cols;
   while (col < endcol) {
     int until = 0;
     for (size_t i = 0; i < kv_size(layers); i++) {
       ScreenGrid *g = kv_A(layers, i);
-      if (g->comp_row > row || row >= g->comp_row + g->Rows
+      grid_rows = g->Rows;
```

Comment:
lets make sure `g->comp_rows` is always valid then we can just use `int grid_rows = MIN(g->Rows, g->comp_rows);`